src/_Common/Quotes/Quotes.Functions.cs

```diff
@@ -80,6 +80,40 @@ public static IEnumerable<TQuote> Validate<TQuote>(this IEnumerable<TQuote> quot
                     });
         }
 
+        public static IEnumerable<Quote> Aggregate<TQuote>(
+            this IEnumerable<TQuote> quotes,
+            TimeSpan newPeriod)
+            where TQuote : IQuote
+        {
+
+            // handle no quotes scenario
+            if (quotes == null || !quotes.Any())
+            {
+                return new List<Quote>();
+            }
+          
+            if (newPeriod == TimeSpan.Zero)
+            {
+                throw new ArgumentOutOfRangeException(nameof(newPeriod), newPeriod,
+                    "Historical quotes Aggregation must use a New Size value of at least " +
```

Comment:
Does `TimeSpan` support `<=` comparators?  If so, we should validate here to make sure it's greater than `TimeSpan.Zero` + adjust the message in this Exception.  I can modify this if you're not comfortable with writing it.